### PR TITLE
Timeline tab improvements

### LINKED ISSUE #903: No information when expanding alert details in the timeline view
When in the timeline view, alert details have no information when expanded. This makes it difficult to drill down into alert testing and see what values were.

### PR DESCRIPTION
**Pull request by captncraig**
Renders template on expanding panels on timeline tab.

fixes #903

### CODE REVIEW

File: cmd/bosun/web/static/partials/config.html [hunk 1]
```diff
@@ -123,7 +123,55 @@
 </ul>
 <div class="tab-content">
 	<div class="tab-pane" ng-class="{active: tab == 'timeline'}">
-		<div ts-alert-history ng-if="tab == 'timeline' && fromDate && toDate"></div>
```
Comment (mjibson) [inline cmd/bosun/web/static/partials/config.html]: Did it make sense to extend ts-alert-history to know about templates instead of duplicating this?

**Comment by captncraig** [comment]
I duplicated it to avoid changing the history view. I suppose I could merge the markup, but half of it would be unused in either view.

**Comment by captncraig** [comment]
That may not be a bad idea

**Comment by mjibson** [comment]
If half is unused, I'm ok keeping it separate.

File: cmd/bosun/web/static/js/config.ts [hunk 2]
```diff
@@ -162,7 +162,7 @@ bosunControllers.controller('ConfigCtrl', ['$scope', '$http', '$location', '$rou
 		});
 	}
 	$scope.scrollTo = (type:string, name:string) => {
-		var searchRegex = new RegExp("^\\s*"+type+"\\s+"+name+"\\s*\\{", "g");
+		var searchRegex = new RegExp("^\\s*"+type+"\\s+"+name, "gm");
```
Comment (mjibson) [inline cmd/bosun/web/static/js/config.ts]: Why add `m` to the regexp flags? Are we expecting multiline inputs?

File: cmd/bosun/web/static/js/config.ts [hunk 3]
```diff
@@ -41,6 +41,7 @@ interface IConfigScope extends IBosunScope {
 	setTemplateGroup: (group:any) => void;
 	scrollToInterval: (v: string) => void;
 	show: (v: any) => void;
+	loadTimelinePanel: (entry:any ,v: any) =>void;
```
Comment (mjibson) [inline cmd/bosun/web/static/js/config.ts]: fix whitespace: `entry: any, v: any) => void`

File: cmd/bosun/web/static/js/config.ts [hunk 4]
```diff
@@ -314,7 +314,33 @@ bosunControllers.controller('ConfigCtrl', ['$scope', '$http', '$location', '$rou
 	$scope.zws = (v: string) => {
 		return v.replace(/([,{}()])/g, '$1\u200b');
 	};
-
+	
+	$scope.loadTimelinePanel = (entry:any,v:any) =>{
```
Comment (mjibson) [inline cmd/bosun/web/static/js/config.ts]: whitespace: `entry:any, v:any) => {`

File: cmd/bosun/web/static/js/config.ts [hunk 5]
```diff
@@ -314,7 +314,33 @@ bosunControllers.controller('ConfigCtrl', ['$scope', '$http', '$location', '$rou
...
+		if (v.doneLoading && !v.error){return;}
```
Comment (mjibson) [inline cmd/bosun/web/static/js/config.ts]: format this like gofmt would

File: cmd/bosun/web/static/js/config.ts [hunk 6]
```diff
@@ -314,7 +314,33 @@ bosunControllers.controller('ConfigCtrl', ['$scope', '$http', '$location', '$rou
...
+		v.error = null;
+		v.doneLoading = false;
+		var ak = entry.key;
+		var openBrack = ak.indexOf("{")
```
Comment (mjibson) [inline cmd/bosun/web/static/js/config.ts]: ; at eol

File: cmd/bosun/web/static/js/config.ts [hunk 7]
```diff
@@ -314,7 +314,33 @@ bosunControllers.controller('ConfigCtrl', ['$scope', '$http', '$location', '$rou
...
+		var closeBrack = ak.indexOf("}")
```
Comment (mjibson) [inline cmd/bosun/web/static/js/config.ts]: ; at eol, same for others

File: cmd/bosun/web/static/js/config.ts [hunk 8]
```diff
@@ -314,7 +314,33 @@ bosunControllers.controller('ConfigCtrl', ['$scope', '$http', '$location', '$rou
...
+		var alertName = ak.substr(0,openBrack)
+		var template = ak.substring(openBrack+1,closeBrack)
```
Comment (mjibson) [inline cmd/bosun/web/static/js/config.ts]: space after comma

File: cmd/bosun/web/static/js/config.ts [hunk 9]
```diff
@@ -314,7 +314,33 @@ bosunControllers.controller('ConfigCtrl', ['$scope', '$http', '$location', '$rou
...
+		var url = '/api/rule?' +
+			'alert=' + encodeURIComponent(alertName) +
+			'&from=' + encodeURIComponent(moment.utc(v.Time).format()) +
+			'&template_group=' + encodeURIComponent(template);
+		$http.post(url,$scope.config_text)
```
Comment (mjibson) [inline cmd/bosun/web/static/js/config.ts]: space after comma

File: cmd/bosun/web/static/js/config.ts [hunk 10]
```diff
@@ -314,7 +314,33 @@ bosunControllers.controller('ConfigCtrl', ['$scope', '$http', '$location', '$rou
...
+			.success((data) => {
+				v.subject = data.Subject;
+				v.body = data.Body;
+			})
+			.error((error) => {
+				v.error = error;
+			})
+			.finally(()=>{
```
Comment (mjibson) [inline cmd/bosun/web/static/js/config.ts]: spaces around `=>`

File: cmd/bosun/web/static/js/directives.ts [hunk 11]
```diff
@@ -218,8 +218,11 @@ bosunApp.directive('tsTimeLine', () => {
 	return {
 		link: (scope: any, elem: any, attrs: any) => {
 			scope.shown = {};
-			scope.collapse = (i: any) => {
+			scope.collapse = (i: any, entry:any, v: any) => {
 				scope.shown[i] = !scope.shown[i];
+				if (scope.loadTimelinePanel && entry && scope.shown[i]){
```
Comment (mjibson) [inline cmd/bosun/web/static/js/directives.ts]: when does `scope.loadTimelinePanel` not exist? why do we need to check for it here?

Comment (captncraig) [inline cmd/bosun/web/static/js/directives.ts]: Because the scope on the history view does not define it, because it needs no lazy-loading.

Comment (mjibson) [inline cmd/bosun/web/static/js/directives.ts]: ohh this is for both of them. yes.

**Pull request merged**